desktop/src/pages/ChatPage/ChatInput/index.tsx

```diff
@@ -17,55 +17,155 @@ import {
   AIInputToolbar,
   AIInputTools,
 } from '@/components/kibo/ai-input';
+// Remove this import - ToolContext is defined below
 import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from '@/components/ui/tooltip';
 import { cn } from '@/lib/utils/tailwind';
-import { useChatStore } from '@/stores/chat-store';
+import { useChatContext } from '@/providers/chat-provider';
+import { useAgentStore } from '@/stores/agent-store';
 import { useDeveloperModeStore } from '@/stores/developer-mode-store';
 import { useMCPServersStore } from '@/stores/mcp-servers-store';
 import { useOllamaStore } from '@/stores/ollama-store';
 import { ChatInteractionStatus } from '@/types';
 
+// Define ToolContext interface
+interface ToolContext {
+  serverName: string;
+  toolName: string;
+  enabled: boolean;
+  description?: string;
+}
```

Comment:
please see `src/types.ts` -- let's avoid having > 1 type definition for things like chat, messages, tools, etc (unless **absolutely** necesary)